### PR TITLE
Sm/ts-update

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
fixes the ts-update errors
uses typescript in more standard ways (reduced `any`, `unknown` only where it's really `unknown`, new types and type guards)
pretty serious general refactoring 
skip 1 test--I'd like some review of that to see if it could be re-enabled and how.  I don't understand the test's `.call` and `context` stuff enough to mess with it.

testing: QA this one much more seriously, and compare file/json outputs to what the current version is doing.  There's big changes, not just the ts fixes that the other 4 repos have.

### What issues does this PR fix or reference?
@W-10054614@

### CODE REVIEW

File: src/api/data/tree/importApi.ts [hunk 1]
```diff
@@ -473,36 +469,32 @@ export class ImportApi {
...
-          const field = (error as Error).message.split("'")[1];
-          const object = (error as Error).message.substr(
-            (error as Error).message.lastIndexOf(' ') + 1,
-            error.message.length
-          );
-          throw SfdxError.create('@salesforce/plugin-data', 'importApi', 'FlsError', [field, object]);
-        }
-        throw SfdxError.wrap(error);
-      });
+    try {
```
Comment (WillieRuemmele) [inline src/api/data/tree/importApi.ts:481]: awesome conversion

File: src/reporters.ts [hunk 2]
```diff
@@ -145,38 +144,35 @@ export class HumanReporter extends QueryReporter {
...
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any
-              childRecords.forEach((record: unknown) => {
-                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-                const newRecord = Object.assign({});
-                Object.entries(record as never).forEach(([key, value]) => {
+          Object.keys(childrenRows).forEach((child) => {
+            const childObject = childrenRows[child];
+            if (childObject && hasNestedRecords<BasicRecord>(childObject)) {
+              childObject.records.forEach((childRecord) => {
+                const newRecord: Record<string, unknown> = {};
```
Comment (WillieRuemmele) [inline src/reporters.ts:164]: loving all of the removed rules

File: src/api/data/tree/exportApi.ts [hunk 3]
```diff
@@ -5,29 +5,24 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-/* eslint-disable  @typescript-eslint/no-explicit-any */
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
```
Comment (WillieRuemmele) [inline src/api/data/tree/exportApi.ts:9]: YES!

**Review by WillieRuemmele — APPROVED** [review]
From my understanding it looks the same, but with a lot more types 😄

**Review by shetzel — APPROVED** [review]

**Pull request merged**